common/src/main/java/org/apache/sedona/common/utils/GeomUtils.java

```diff
@@ -254,7 +254,17 @@ public static Geometry transform(Geometry geometry, CoordinateReferenceSystem so
     }
 
     public static int getDimension(Geometry geometry) {
-        return geometry.getCoordinate() != null && !java.lang.Double.isNaN(geometry.getCoordinate().getZ()) ? 3 : 2;
+        int dimension = 2;
```

Comment:
This function is duplicate to nDims(): https://github.com/apache/sedona/blob/master/common/src/main/java/org/apache/sedona/common/Functions.java#L298

Please copy the logic of nDims here and let nDims function call this getDimension function.

Reply:
Got it.